packages/flutter/lib/src/widgets/animated_scroll_view.dart

```diff
@@ -211,6 +211,341 @@ class AnimatedListState extends _AnimatedScrollViewState<AnimatedList> {
+///       return const Placeholder();
+///     },
+///     separatorBuilder: (BuildContext context, int index, Animation<double> animation) {
+///       return const Placeholder();
+///     },
+///   );
+/// }
+///
+/// // ...
+///
+/// void _updateList() {
+///   // adds "123" to the AnimatedListSeparated
+///   listKey.currentState!.insertItem(123);
+/// }
+/// ```
+///
+/// [AnimatedListSeparated] item input handlers can also refer to their [AnimatedListSeparatedState]
+/// with the static [AnimatedListSeparated.of] method.
+class AnimatedListSeparatedState extends _AnimatedScrollViewState<AnimatedListSeparated> {
+
+  @override
+  Widget build(BuildContext context) {
+    return _wrap(
+      SliverAnimatedList(
+        key: _sliverAnimatedMultiBoxKey,
+        itemBuilder: widget.itemBuilder,
+        initialItemCount: widget.initialItemCount,
+      ),
+      widget.scrollDirection,
+    );
+  }
+
+  /// Insert an item at [index] with a corresponding separator
+  /// and start an animation that will be passed to
+  /// [AnimatedListSeparated.itemBuilder] and [AnimatedListSeparated.separatorBuilder]
+  /// when the item is visible.
+  ///
+  /// This method's semantics are the same as Dart's [List.insert] method: it
+  /// increases the length of the list of items by one and shifts
+  /// all items at or after [index] towards the end of the list of items.
+  @override
+  void insertItem(int index, { Duration duration = _kDuration }) {
+      final int itemIndex = _computeItemIndex(index);
+      super.insertItem(itemIndex, duration: duration);
+      if (_itemsCount > 1) {
+        super.insertItem(itemIndex, duration: duration);
+      }
+  }
+
+  /// Insert multiple items at [index] with corresponding separators
+  /// and start an animation that will be passed to
+  /// [AnimatedListSeparated.itemBuilder] and [AnimatedListSeparated.separatorBuilder]
+  /// when the items are visible.
+  @override
+  void insertAllItems(int index, int length, { Duration duration = _kDuration, bool isAsync = false }) {
+    final int itemIndex = _computeItemIndex(index);
+    final int lengthWithSeparators = _itemsCount == 0 ? length * 2 - 1 : length * 2;
+    super.insertAllItems(itemIndex, lengthWithSeparators, duration: duration, isAsync: isAsync);
+  }
+
+  /// This method is not supported by [AnimatedListSeparatedState]. Use [removeSeparatedItem] instead.
+  @override
+  void removeItem(int index, AnimatedRemovedItemBuilder builder, { Duration duration = _kDuration }) {
+    throw Exception('Separated list items can not be removed with this method. Use removeSeparatedItem instead.');
+  }
+
+  /// This method is not supported by [AnimatedListSeparatedState]. Use [removeAllSeparatedItems] instead.
+  @override
+  void removeAllItems(AnimatedRemovedItemBuilder builder, { Duration duration = _kDuration }) {
+    throw Exception('Separated list items can not be removed with this method. Use removeAllSeparatedItems instead.');
+  }
```

Comment:
I think this would lead to confusion. Can we instead find some way to let `removeItem` be how you remove items, for consistency across the various APIs?

Reply:
e.g. have the superclass have an `@protected` `internalRemoveItem` or some such, and then have the subclasses other than this one introduce a `removeItem` that just forwards everything to `internalRemoveItem`?